File: modulo 2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int spaces = 0, frases = 0, n, i, words = 0;
    float l;

    string text = get_string("Digite o texto: ");

    for (i = 0, n = (strlen(text)); i <= n; i++) // contagem de letras, palavras e frases
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122)) // letras
        {
            words++;
        }
        if (text[i] == 32) //palavras
        {
            spaces++;

        }
        if (text[i] == 46 || text[i] == 33 || text[i] == 63) // frases
        {
            frases++;

        }
    }
    spaces = spaces + 1; //correção de bugs
    printf("Numero de letras: %i\n",  words);
    printf("Numero de palavras: %i\n", spaces);
    printf("Numero de sentenças: %i\n", frases);
    l = ((0.0588 * (((float) words / spaces) * 100) - (0.296 * (((float) frases / spaces) * 100)) - 15.8)); // indice de coleman-liau
    if (l >= 16) //condições para mostrar o resultado
    {
        printf("\nGrade 16+\n");
    }
    else if (l < 1)
    {
        printf("\nBefore Grade 1\n");
    }
    else
    {
        printf("\nGrade %.0f\n", round(l));
    }


}

File: modulo 6/readability.py
from cs50 import get_string

text = get_string("Text: ")

# Contagem de letras
n_letras = len([letra for letra in text if letra.isalpha()])


# Contagem de palavras
n_palavras = len(text.split())

# Contagem de frases
n_frases = text.count('.')+text.count('!')+text.count('?')

L = n_letras * 100 / n_palavras
S = n_frases * 100 / n_palavras

index = (0.0588 * L) - (0.296 * S) - 15.8

print("Número de letras:", n_letras)
print("Número de palavras:", n_palavras)
print("Número de frases:", n_frases)
print("")

if round(index) > 16:
    print("Grade 16+")
elif round(index) < 1:
    print("Before Grade 1")
else:
    print("Grade", round(index))
